### ISSUE TITLE
etcd does not survive restarts

### ISSUE THREAD

**Issue by ryane** [CiscoCloud/microservices-infrastructure#737]
etcd does not survive restarts

If etcd is restarted or if a node is rebooted, etcd cannot start:

```
Oct 14 20:45:12 ip-10-0-175-75.ec2.internal systemd[1]: Started etcd.
Oct 14 20:45:12 ip-10-0-175-75.ec2.internal docker[30348]: 2015/10/14 20:45:12 etcdmain: setting maximum number of CPUs to 1, total number of available CPUs is 4
Oct 14 20:45:12 ip-10-0-175-75.ec2.internal docker[30348]: 2015/10/14 20:45:12 etcdmain: no data-dir provided, using default data-dir ./resching-aws-worker-002.etcd
Oct 14 20:45:12 ip-10-0-175-75.ec2.internal docker[30348]: 2015/10/14 20:45:12 etcdmain: listening for peers on http://0.0.0.0:2380
Oct 14 20:45:12 ip-10-0-175-75.ec2.internal docker[30348]: 2015/10/14 20:45:12 etcdmain: listening for client requests on http://0.0.0.0:2379
Oct 14 20:45:12 ip-10-0-175-75.ec2.internal docker[30348]: 2015/10/14 20:45:12 etcdmain: stopping listening for client requests on http://0.0.0.0:2379
Oct 14 20:45:12 ip-10-0-175-75.ec2.internal docker[30348]: 2015/10/14 20:45:12 etcdmain: stopping listening for peers on http://0.0.0.0:2380
Oct 14 20:45:12 ip-10-0-175-75.ec2.internal docker[30348]: 2015/10/14 20:45:12 etcdmain: member 3086be9801215576 has already been bootstrapped
Oct 14 20:45:12 ip-10-0-175-75.ec2.internal systemd[1]: etcd.service: main process exited, code=exited, status=1/FAILURE
```

With the current configuration, it looks like etcd starts up with a new, empty data-dir so I suspect that is a big part of the problem. We probably need to mount a volume for the data-dir but not sure what else might be needed.

ping @loshkovskyi

**Comment by stevendborrelli** [comment]
Due to this issue, with #738 we've commented out this feature.

**Comment by loshkovskyi** [comment]
I confirm the issue is related to the cluster state `--initial-cluster-state new`. It should be changed to `--initial-cluster-state existing` when an `etcd` node is restarted in order to join the cluster correctly.

A possible solution can be as following:
1. Bootstrap `etcd` cluster with `--initial-cluster-state new` for each node.
2. Update [systemd unit file](https://github.com/CiscoCloud/microservices-infrastructure/blob/master/roles/etcd/templates/etcd.service.j2) with `--initial-cluster-state existing`.
3. Reload `systemd` daemon.

**Comment by Zogg** [comment]
Maybe adding 'sed' line in `ExecStartPost` of the etcd's systemd config file would do the trick.

**Comment by avnik** [comment]
@zogg  cherry-picked commits with fix to his k8s branch.
9ccdeb4174401346d8eeabb366aaf9f5dd63795c I used wrapper script to strip `INITIAL_*` variables from env, and switch INITIAL_CLUSTER_STATE to existng.
f388586e39bba722e1d24ff69470595631b44323 Also I removed `localhost` from initial cluster, which are cause funny effects like "can't find self in cluster"

This issue can be closed when @zogg's branch will be merged

**Comment by ryane** [comment]
fixed by #794
